libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/camera/view/MapboxRecenterButton.kt

```diff
@@ -0,0 +1,144 @@
+import com.mapbox.navigation.ui.utils.internal.extensions.shrink
+import com.mapbox.navigation.ui.utils.internal.extensions.slideWidth
+
+/**
+ * Default view to allow user to switch to route overview mode.
+ */
+class MapboxRecenterButton : ConstraintLayout {
+
+    private var textWidth = 0
+    private var isAnimationRunning = false
+    private val mainHandler = Handler(Looper.getMainLooper())
+    private val binding = MapboxRecenterLayoutBinding.inflate(
+        LayoutInflater.from(context),
+        this,
+        true
+    )
+
+    /**
+     *
+     * @param context Context
+     * @constructor
+     */
+    constructor(context: Context) : super(context)
+
+    /**
+     *
+     * @param context Context
+     * @param attrs AttributeSet?
+     * @constructor
+     */
+    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs) {
+        initAttributes(attrs)
+    }
+
+    /**
+     *
+     * @param context Context
+     * @param attrs AttributeSet?
+     * @param defStyleAttr Int
+     * @constructor
+     */
+    constructor(
+        context: Context,
+        attrs: AttributeSet?,
+        defStyleAttr: Int
+    ) : super(context, attrs, defStyleAttr) {
+        initAttributes(attrs)
+    }
+
+    override fun onFinishInflate() {
+        super.onFinishInflate()
+        binding.recenterText.afterMeasured {
+            textWidth = width
+        }
+    }
+
+    /**
+     * Allows you to change the style of [MapboxRecenterButton].
+     * @param style Int
+     */
+    fun updateStyle(@StyleRes style: Int) {
+        val typedArray = context.obtainStyledAttributes(
+            style,
+            R.styleable.MapboxRecenterButton
+        )
+        applyAttributes(typedArray)
+        typedArray.recycle()
+    }
+
+    /**
+     * Invoke the function to show optional text associated with the view.
+     * @param duration for the view to be in the extended mode before it starts to shrink.
+     */
+    fun showTextAndExtend(duration: Long) {
+        if (!isAnimationRunning) {
+            isAnimationRunning = true
+            val extendToWidth = EXTEND_TO_WIDTH * context.resources.displayMetrics.density
+            val animator = getAnimator(textWidth, extendToWidth.toInt())
+            binding.recenterText.extend(
+                animator,
+                doOnStart = {
+                    binding.recenterText.text = context.getString(R.string.mapbox_recenter)
+                    binding.recenterText.visibility = View.VISIBLE
+                    mainHandler.postDelayed(
+                        {
+                            val endAnimator = getAnimator(extendToWidth.toInt(), textWidth)
+                            binding.recenterText.shrink(
+                                endAnimator,
+                                doOnStart = {
+                                    binding.recenterText.text = null
+                                    isAnimationRunning = false
```

Comment:
`isAnimationRunning = false` should be moved under `doOnEnd`. Here and below(`MapboxRouteOverviewButton.kt `)